File: AOJ/0360/0360.py
def overlap(a, b, c, d):
    if d <= a:  return False
    if b <= c:  return False
    return True

a, b = map(int, input().split())
res = 0
for i in range(int(input())):
    s, t = map(int, input().split())
    if overlap(a, b, s, t): res = 1
print(res)

File: AOJ/0360/0360.c
#include<stdio.h>

int overlap(int a, int b, int c, int d){
  if ( d <= a ) return 0;
  if ( b <= c ) return 0;
  return 1;
}

int main(){
  int a, b, s, f, n, i;
  scanf("%d %d %d", &a, &b, &n);
  for ( int i = 0; i < n; i++ ){
    scanf("%d %d", &s, &f);
    if ( overlap(a, b, s, f) ){
      printf("1\n");
      return 0;
    }
  }
  printf("0\n");
  return 0;
}

